platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -245,291 +518,180 @@ message RoundRosterPair {
-PCES event version matches the current software version, the active roster is used. This behavior will be
-maintained until Birth Rounds or Dynamic Address Book is implemented.
+    /**
+     * Gets the roster history.
+     * Returns the active roster history iff:
+     *      the roster state singleton is not null
+     *      the list of round roster pairs is not empty
+     *      the active roster hashes are present in the roster map
+     * otherwise returns null.
+     * @return the active rosters
+     */
+    @Nullable
+    List<RoundRosterPair> getRosterHistory();
 
-Therefore, Provided an existing State, an `Override Roster` and a network upgrade (which could be a simple
-config-only upgrade if a software version upgrade isn't required), the platform will adopt the provided Override Roster.
+    /**
+     * Get the roster based on roster hash
+     *
+     * @param rosterHash The roster hash
+     * @return The roster.
+     */
+    @Nullable
+    Roster get(@NonNull Bytes rosterHash);
 
-The next round number for this network will continue from the last round number in the provided state.
-The trigger for adopting the `Override Roster` will be the presence of an Override Roster at start-up with a State
-existing on disk. Bear in mind the Network Transplant process is designed primarily with test networks in mind.
-Upon adoption of the Override Roster, the Override Roster should be deleted from disk. In test networks where this
-process is designed to be used, the Override Roster can be easily accessed and deleted by DevOps.
+}
+```
 
-A new method will be added to the `PlatformBuilder` that will be used by the App to set the Override Roster. In most
-cases, as it's the case for a Genesis Roster, there will be no need to call the method on `PlatformBuilder` to set
-the Override Roster except in the events we need to transplant a network.
+##### WritableRosterStore
 
 ```java
-//in PlatformBuilder
-/*
-  Set the Override Roster for the network.
-  <p>
-  This method is used to set the optional Override Roster for the network.
-  The Override Roster is a special roster used for the sole purpose of Transplanting an existing network.
-  The Override Roster, if set, is immutable and will be used to trigger a Network Transplant process.
-  <p>
-  @param overrideRoster the Override Roster for the network.
+/**
+ * Read-write implementation for accessing rosters states.
  */
-PlatformBuilder withOverrideRoster(@NonNull final Roster overrideRoster);
-```
+public class WritableRosterStore extends ReadableRosterStore {
 
-#### New Transplant Procedure
+    /**
+     * Sets the candidate roster in state.
+     * Setting the candidate roster indicates that this roster should be adopted as the active roster when required.
+     *
+     * @param candidateRoster a candidate roster to set. It must be a valid roster.
+     */
+    public void putCandidateRoster(@NonNull final Roster candidateRoster);
```

Comment:
Why `putCandidateRoster()` and not `setCandidateRoster()`?

Reply:
@derektriley  will have to explain that one. :-). https://github.com/hashgraph/hedera-services/pull/15442/commits/9771b715022758d8fb75ed420dac885f7e9f6191